proposals/1.2-proposal.md

```diff
@@ -0,0 +1,113 @@
+    - If wrap mode in on - Max length on line 80 chars
+
+```
+-----------------------------------------------------------------------------
+~Version Information Section
+VERS.             1.2:   CWLS log ASCII Standard -VERSION 1.2
+WRAP.               NO:   One line per depth step
+-----------------------------------------------------------------------------
+```
+**Proposed Data struct - Dictionary**
+
+~W (Well Information)
+
+- This section is **mandatory**.
+- It identifies the well, its location and indicates the start and stop depths of the file.Basically every information that is there for the description of well
+- This segment contains a lot of **mnemonics**
+```
+-----------------------------------------------------------------------------
+~Well Information Section
+#MNEM.UNIT	Data Type	Information  
+#---------	-------------	------------------------------
+STRT.M           635.0000:
+STOP.M           400.0000:
+STEP.M            -0.1250:
+NULL.             -999.25:
+COMP.             COMPANY:      ANY OIL COMPANY INC.
+WELL.                WELL:      ANY ET AL A9-16-49-20
+FLD .               FIELD:      EDAM
+LOC .            LOCATION:      A9-16-49-20W3M
+PROV.            PROVINCE:      SASKATCHEWAN
+SRVC.     SERVICE COMPANY:      ANY LOGGING COMPANY INC.
+DATE.            LOG DATE:      13-DEC-86
+UWI .      UNIQUE WELL ID:      100091604920W300
+-----------------------------------------------------------------------------
+```
+**Proposed Data Struct- Table with MNEM, unit data type and Info as columns**
+
+~C (Curve Information)
+- This basically contains all type of measurements of the particular well that were taken, ie, contains the list of all the possible values that are there in the data table.
+- By possible values, it is meant that type of measurements like - Depth, Bulk Density, etc.
+- The curves described in this section must be present in the data set.
+```
+-----------------------------------------------------------------------------
+~Curve Information Section
+#MNEM.UNIT	API CODE	Curve Description	
+#---------    -------------   -------------------------------
+DEPTH.M                      :    1       DEPTH
+RHOB .K/M3        7 350 02 00:    2       BULK DENSITY
+NPHI .VOL/VOL     7 890 00 00:    3       NEUTRON POROSITY - SANDSTONE
+MSFL .OHMM        7 220 01 00:    4       Rxo RESISTIVITY
+SFLA .OHMM        7 222 01 00:    5       SHALLOW RESISTIVITY
+ILM  .OHMM        7 120 44 00:    6       MEDIUM RESISTIVITY
+ILD  .OHMM        7 120 46 00:    7       DEEP RESISTIVITY
+SP   .MV          7 010 01 00:    8       SPONTANEOUS POTENTIAL
+GR   .GAPI        7 310 01 00:    9       GAMMA RAY
+CALI .MM          7 280 01 00:    10      CALIPER
+DRHO .K/M3        7 356 01 00:    11      DENSITY CORRECTION
+-----------------------------------------------------------------------------
+```
+**Proposed Data Struct - Table with MNEM, unit, API and Curve description as columns**
```

Comment:
I would use a dictionary again to facilitate queries like `curveinfo[:SP][:UNIT]` or `curveinfo[:SP][:DESC]`